mlptrain/potentials/mace/mace.py

```diff
@@ -0,0 +1,583 @@
+                                      'here: https://github.com/ACEsuit/mace')
+            
+        _, all_train_configs = data.load_from_xyz(file_path=train_path,
+                                                  config_type_weights=config_type_weights,
+                                                  energy_key=energy_key,
+                                                  forces_key=forces_key,
+                                                  extract_atomic_energies=False)
+        logging.info(f"Loaded {len(all_train_configs)} training configurations from '{train_path}'")
+
+        if valid_path is not None:
+            _, valid_configs = data.load_from_xyz(
+            file_path=valid_path,
+            config_type_weights=config_type_weights,
+            energy_key=energy_key,
+            forces_key=forces_key,
+            extract_atomic_energies=False)
+            logging.info(
+            f"Loaded {len(valid_configs)} validation configurations from '{valid_path}'")
+            train_configs = all_train_configs
+        else:
+            logging.info(
+            "Using random %s%% of training set for validation", 100 * valid_fraction)
+            train_configs, valid_configs = data.random_train_valid_split(
+            all_train_configs, valid_fraction, seed)
+
+        return train_configs, valid_configs
+    
+    def create_error_table( self, table_type, all_collections, z_table, r_max, valid_batch_size, model, loss_fn, device):
+        table = PrettyTable()
+        
+        try:
+            from mace import data
+            from mace.tools import torch_geometric,  evaluate
+
+        except ModuleNotFoundError:
+            raise ModuleNotFoundError('MACE install not found, install it '
+                                      'here: https://github.com/ACEsuit/mace')
+            
+        if table_type == "TotalRMSE":
+            table.field_names = [
+            "config_type",
+            "RMSE E / meV",
+            "RMSE F / meV / A",
+            "relative F RMSE %",
+            ]
+        elif table_type == "PerAtomRMSE":
+            table.field_names = [
+            "config_type",
+            "RMSE E / meV / atom",
+            "RMSE F / meV / A",
+            "relative F RMSE %",
+            ]
+        elif table_type == "TotalMAE":
+            table.field_names = [
+            "config_type",
+            "MAE E / meV",
+            "MAE F / meV / A",
+            "relative F MAE %",
+            ]
+        elif table_type == "PerAtomMAE":
+            table.field_names = [
+            "config_type",
+            "MAE E / meV / atom",
+            "MAE F / meV / A",
+            "relative F MAE %",
+            ]
+        for name, subset in all_collections:
+            data_loader = torch_geometric.dataloader.DataLoader(
+            dataset=[
+                data.AtomicData.from_config(config, z_table=z_table, cutoff=r_max)
```

Comment:
Consider for loops without concatenation if otherwise it requires going over multiple lines